pkg/global/vars/vars.go

```diff
@@ -0,0 +1,72 @@
+package vars
+
+import (
+	"os"
+	"regexp"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/global/consts"
+)
+
+var (
```

Comment:
after reviewing all commits (its a huge PR so i trust we just moved things around and defined interfaces and not applied any logic changes) i wonder if using vars is the correct way to go.

personally i think it makes individual components hard to understand as their behaviour depends on some external package.

i believe the correct way to go is to try to eliminate the use of global variables and either have the appropriate interface expose the information (set during interface instance creation) and also pass in relevant attributes (e.g nodeName) to relevant components.

that said, one could argue that it was like that before and this commit just moved it to a single place.
so im not sure what to think of it.

Reply:
as I commented above about this one, the global vars is for configuration that is used in different part of the code but they are global meaning they don't change in the run for example cluster type or development flag.

I really try to reduce passing the same info that will never change down the interfaces.
for example the systemdMode it was like this before the change
main -> daemon -> plugins -> utils -> chroot

and I think it's really not useful to pass that boolean if it's really global if we are in systemd it will be the case for the all daemon run

Reply:
not really a fan of the approach, but i get it.
in any case this PR improves the code base.